src/JustSaying/Fluent/AccountAddressProvider.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using Amazon;
+using JustSaying.Naming;
+
+namespace JustSaying.Fluent
+{
+    /// <summary>
+    /// Creates resource addresses either by name or naming convention.
+    /// </summary>
+    public sealed class AccountAddressProvider
+    {
+        private readonly string _accountId;
+        private readonly RegionEndpoint _regionEndpoint;
+        private readonly IQueueNamingConvention _queueNamingConvention;
+        private readonly ITopicNamingConvention _topicNamingConvention;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AccountAddressProvider"/> class with the default naming convention.
+        /// </summary>
+        /// <param name="accountId">The AWS account ID the topics and queues belong in.</param>
+        /// <param name="regionName">The AWS region the topics and queues belong in.</param>
+        public AccountAddressProvider(string accountId, string regionName)
```

Comment:
Just an idea (as long as the public API surface area doesn't explode with possibilities), maybe also expose the ability to directly pass in a `RegionEndpoint`?